.github/update_grid_driver.yaml

```diff
@@ -0,0 +1,49 @@
+name: Update NVIDIA Drivers
+
+on:
+  schedule:
+    - cron: '0 0 * * *'  # Runs at 00:00 UTC every day
+  workflow_dispatch:  # Allows manual trigger
+
+jobs:
+  update-drivers:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install ruamel.yaml requests
```

Comment:
no fixed version ? 
you should be using fixed version and have Dependabot open PR against those dependency (use requirements.txt)

Also, I"m not sure for GitHub but in OneBranch we can only pull from ADO Artifacts (not public) this might be coming over GitHub.

Reply:
Added the requirements.txt.

Reply:
Is there an ADO mirror of pypi artifacts, that we can pull from? I'll search and see if I can find one.

Reply:
your pipeline doesn't run on ADO though . so Public pulling is fine for now.